tree/tree/src/TBranch.cxx

```diff
@@ -2036,6 +2047,22 @@ TBranch* TBranch::GetMother() const
 {
    if (fMother) return fMother;
 
+   {
+      TBranch *parent = fParent;
+      while(parent) {
+         if (parent->fMother) {
+            const_cast<TBranch*>(this)->fMother = parent->fMother; // We can not yet use the 'mutable' keyword
+            return fMother;
+         }
+         if (!parent->fParent) {
+            // This is the top node
+            const_cast<TBranch*>(this)->fMother = parent; // We can not yet use the 'mutable' keyword
+            return fMother;
+         }
+         parent = parent->fParent;
+      }
+   }
```

Comment:
This code is quite confusing. Just so I understand, what is the difference between `fParent` and `fMother` here? And why can't we use 'mutable'?

Reply:
The `only` reason to not use `mutable` is/was to reduce the code churn.  When the (similar) code was written `mutable` was not working properly on all platforms.   It can now be used and the code 'simplified'.

`fParent` is the immediate parent.
`fMother` is the top level parent (the ultimate parent, one of the top level branches)